module/VuFind/src/VuFind/Content/Covers/Bokinfo.php

```diff
@@ -0,0 +1,200 @@
+<?php
+
+/**
+ * VuFind Driver for Koha, using REST API
+ *
+ * PHP version 7
+ *
+ * Copyright (C) The National Library of Finland 2016-2020.
+ * Copyright (C) Moravian Library 2019.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Content
+ * @author   Jacob Sandin <jacob@imcode.com>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     Wiki
+ * https://vufind.org/wiki/development:plugins:content_provider_components#covers
+ */
+namespace VuFind\Content\Covers;
+
+use SimpleXMLElement;
+
+/**
+ * Summon cover content loader.
+ *
+ * @category VuFind
+ * @package  Content
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class Bokinfo extends \VuFind\Content\AbstractCover implements
+    \Laminas\Log\LoggerAwareInterface,
+    \VuFindHttp\HttpServiceAwareInterface
+{
+    use \VuFind\Log\LoggerAwareTrait;
+    use \VuFindHttp\HttpServiceAwareTrait;
+
+    /**
+     * Constructor
+     */
+    public function __construct()
+    {
+        $this->supportsIsbn = $this->cacheAllowed = true;
```

Comment:
Just to be safe: have you confirmed that the Bokinfo terms of service allow caching?

Reply:
No I have not, but I will send a question right away.

Reply:
Thank you! It's always good to check this point because different vendors have different opinions. Some prefer us to cache locally so we reduce load on their servers; others don't allow caching because of copyright concerns.